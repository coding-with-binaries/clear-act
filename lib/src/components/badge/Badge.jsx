import PropTypes from 'prop-types';
import React from 'react';
import cn from 'classnames';

export const Badge = React.forwardRef((props, ref) => {
  const { badgeType, children, ...attrs } = props;

  const className = cn(props.className, 'badge', {
    [`badge-${badgeType}`]: !!badgeType
  });

  return (
    <span ref={ref} className={className} {...attrs}>
      {children}
    </span>
  );
});

Badge.propTypes = {
  badgeType: PropTypes.string
};
