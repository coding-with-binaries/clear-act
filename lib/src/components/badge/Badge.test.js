import { Badge } from './Badge';

import React from 'react';
import renderer from 'react-test-renderer';

test('Renders Basic badge', () => {
  const component = renderer.create(<Badge>Basic Badge</Badge>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Status badges', () => {
  const component = renderer.create(
    <>
      <Badge badgeType="info">Info Badge</Badge>
      <Badge badgeType="success">Success Badge</Badge>
      <Badge badgeType="warning">Warning Badge</Badge>
      <Badge badgeType="danger">Danger Badge</Badge>
    </>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
