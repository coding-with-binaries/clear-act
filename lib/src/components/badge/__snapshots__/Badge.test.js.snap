// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Basic badge 1`] = `
<span
  className="badge"
>
  Basic Badge
</span>
`;

exports[`Renders Status badges 1`] = `
Array [
  <span
    className="badge badge-info"
  >
    Info Badge
  </span>,
  <span
    className="badge badge-success"
  >
    Success Badge
  </span>,
  <span
    className="badge badge-warning"
  >
    Warning Badge
  </span>,
  <span
    className="badge badge-danger"
  >
    Danger Badge
  </span>,
]
`;
